packages/generator/src/v1.1/generatePerActionDataFiles.ts

```diff
@@ -18,41 +18,136 @@ import type { ActionParameterType, ActionTypeV2 } from "@osdk/gateway/types";
+import { getDescriptionIfPresent } from "./wireObjectTypeV2ToV1ObjectInterfaceString";
+
+function stringifyWithoutOutterBraces(obj: any) {
+  return JSON.stringify(obj, null, 2).replace(/^\{\n/, "").replace(/\n\}$/, "");
+}
 
 export async function generatePerActionDataFiles(
   ontology: WireOntologyDefinition,
   fs: MinimalFs,
   outDir: string,
-  importExt: string = "",
+  importExt: string,
+  v2: boolean,
 ) {
   await fs.mkdir(outDir, { recursive: true });
   await Promise.all(
     Object.values(ontology.actionTypes).map(async (action) => {
       const uniqueApiNames = new Set(
         extractReferencedObjectsFromAction(action),
       );
+
+      const uniqueApiNamesString = uniqueApiNames.size > 0
+        ? [...uniqueApiNames].map(apiName => `"${apiName}"`).join("|")
+        : "never";
+
+      const fullActionDef = deleteUndefineds(
+        wireActionTypeV2ToSdkActionDefinition(action),
+      );
+
+      const { parameters, ...actionDefSansParameters } = fullActionDef;
+
+      const actionDefIdentifier = `ActionDef$${action.apiName}`;
+      const paramsDefIdentifier = `${actionDefIdentifier}$Params`;
+      const paramsIdentifier = `${action.apiName}$Params`;
+
+      function createInterface() {
+        // the params must be a `type` to align properly with the `ActionDefinition` interface
+        // this way we can generate a strict type for the function itself and reference it from the Aciton Definition
+        return `
+        
+          // Represents the definition of the parameters for the action
+          export type ${paramsDefIdentifier} = {
+            ${
+          Object.entries(parameters)
+            .map(([key, value]) => {
+              const { type, ...remain } = value;
+
+              let q;
+
+              if (typeof type === "string") {
+                q = JSON.stringify(type);
+              } else if (type.type === "object") {
+                q = `ObjectActionDataType<"${type.object}", ${type.object}Def>`;
+              } else if (type.type === "objectSet") {
+                q =
+                  `ObjectSetActionDataType<"${type.objectSet}", ${type.objectSet}Def>`;
+              }
+
+              return `"${key}": {
+                type: ${q};
+                ${stringifyWithoutOutterBraces(remain)}
+              }
+              `;
+            })
+            .join(";\n")
+        }
+          }
+
+          // Represents the runtime arguments for the action
+          export type ${paramsIdentifier} = NOOP<OsdkActionParameters<${paramsDefIdentifier}>>;
+
+          
+          // Represents a fqn of the action
+          export interface ${action.apiName} {
+            ${getDescriptionIfPresent(action.description)}
+             <OP extends ApplyActionOptions>(args: ${paramsIdentifier}, options?: OP): Promise<ActionReturnTypeForOptions<OP>>;
+          }
+
+          
+          // Represents the definition of the action
```

Comment:
I did this intentionally for people who click through. I didn't put them as javadoc because I thought it would confuse things with when we put in descriptions.